CSharp/Library/Microsoft.Bot.Connector.Shared/Authentication/RetryParams.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Bot.Connector.Shared.Authentication
+{
+    public class RetryParams
+    {
+        private static readonly TimeSpan MaxDelay = TimeSpan.FromSeconds(10);
+        private static readonly TimeSpan DefaultBackOffTime = TimeSpan.FromMilliseconds(50);
+
+        public static RetryParams StopRetrying
+        {
+            get
+            {
+                return new RetryParams() { ShouldRetry = false };
```

Comment:
This property is called every time the retry logic kicks in. Allocating a new object seems like unnecessary overhead given that it's used immutably.

Reply:
Ya, yes I had that in mind and never came back to it. Fixing now.

Reply:
Changed it to `public static RetryParams StopRetrying { get; } = new RetryParams() { ShouldRetry = false };` so it gets created once.